rotkehlchen/globaldb/cache.py

```diff
@@ -29,37 +35,47 @@ def globaldb_set_general_cache_values_at_ts(
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it."""
     cache_key = _compute_cache_key(key_parts)
+    unique_cache = any(cache_key.startswith(key_part) for key_part in UNIQUE_CACHE_KEYS)
+    if unique_cache:
+        cache = UNIQUE_CACHE
+    else:
+        cache = GENERAL_CACHE
```

Comment:
since you check `startswith` it means that `GeneralCacheType` is the first part of the cache so you can change it and make `UNIQUE_CACHE_KEYS` a `set[GeneralCacheType]` and check `unique_cache = key_parts[0] in UNIQUE_CACHE_KEYS`